x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsPersister.java

```diff
@@ -218,6 +218,7 @@ public void executeRequest() {
                 BulkResponse addRecordsResponse = client.bulk(bulkRequest).actionGet();
                 if (addRecordsResponse.hasFailures()) {
                     logger.error("[{}] Bulk index of results has errors: {}", jobId, addRecordsResponse.buildFailureMessage());
+                    throw new BulkIndexException(addRecordsResponse);
```

Comment:
I thought it best to throw here, and handle retries up the stack. That way the retries know about the processor state and can stop retrying if the processor died (or is dying).